.github/workflows/npm-publish.yml

```diff
@@ -1,29 +1,35 @@
-# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
-# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages
-
 name: Publish to Npm
 
-on: push
+on:
+  workflow_run:
+    workflows: ["ESLint", "Typecheck", "Test"]
+    types:
+      - completed
+  push:
+    branches:
+      - master
 
 jobs:
   publish:
     runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
     steps:
       - name: Checkout code
-        uses: actions/checkout@v1
+        uses: actions/checkout@v2
+
       - name: Install Node
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v2
         with:
           node-version: 14
-      - name: Install yarn
-        run: npm install -g yarn
+          cache: "yarn"
+
       - name: Install packages
         run: yarn install
```

Comment:
Since we are using yarn and maintaing yarn.lock file. My opinion would be to make it more strict

```
{ 
  ...
  "engines": {
    "npm": "please-use-yarn",
    "yarn": ">= 1.19.1"
  },
  ...
}
```

Reply:
You mean the yarn version? I coulnd't find in the docs how to use `actions/setup-node` with another yarn version.

I would not prefer the `engines` keyword in `package.json`. It should other developers warn when they are installing the package with a version which is not listed. See the docs:

https://docs.npmjs.com/cli/v8/configuring-npm/package-json#engines

To garantie different `node` versions to work, we should also update the test workflow to include a node-version matrix.
But I have no idea how to test different `yarn` versions :')